### PR TITLE
feat(dynamic-flows): Add a displayStepTitle prop to DynamicFlow

### PR DESCRIPTION
**Pull request by JulianG-TW**
## Context

We need a way to tell `DynamicFlow` NOT TO inject the step title into the layout, in some cases. Otherwise, when we render a DF flow in a modal body, we get the title duplicated once in the modal header and once in the modal body:

image

## Changes

We add a new `displayStepTitle` prop to `DynamicFlow`.
This prop is optional and when omitted it defaults to `true`, to maintain backwards compatibility.

## Checklist

- [x] Make PR title meaningful and follow [the commit lint format](https://github.com/transferwise/neptune-web/blob/master/CONTRIBUTING.md#versioning-and-commit-lint) (it will appear in the changelog)
- [x] Changes are tested and all tests pass
- [x] Changes meet [accessibility standards](https://github.com/transferwise/neptune-web/blob/main/ACCESSIBILITY.md) and there are no violations in the console
- [x] Changes work in all supported browsers (don't forget IE11)
- [x] You've updated the documentation if necessary

### CODE REVIEW

File: packages/dynamic-flows/src/step/layoutStep/LayoutStep.js [hunk 1]
```diff
@@ -44,6 +45,7 @@ const LayoutStep = (props) => {
 
 LayoutStep.propTypes = {
   stepSpecification: PropTypes.object.isRequired,
+  stepConversionOptions: PropTypes.object.isRequired,
```
Comment (richardbata) [inline packages/dynamic-flows/src/step/layoutStep/LayoutStep.js:48]: One of those examples where we can rewrite file to TS, if not too much effort 👍

File: packages/dynamic-flows/src/step/layoutStep/LayoutStep.js [hunk 2]
```diff
@@ -6,12 +6,12 @@ import DynamicLayout from '../../layout';
 
 import { convertStepToLayout, inlineReferences } from './layoutService';
 
-const getComponents = (step) => {
+const getComponents = (step, stepConversionOptions) => {
```
Comment (richardbata) [inline packages/dynamic-flows/src/step/layoutStep/LayoutStep.js:9]: Name param just: options?

**Review by richardbata — APPROVED** [review]

File: packages/dynamic-flows/src/step/layoutStep/LayoutStep.js (hunk 1 above)
Comment (JulianG-TW) [reply packages/dynamic-flows/src/step/layoutStep/LayoutStep.js:48]: I was about to do it. I even renamed the file. But then the changes to the file get lost in the diff.
I'm more inclined to make a PR exclusively about JS->TS.

I'm planning to bring some TS types that I wrote based on the DFv3 spec into this repo and change a bunch of files to TS without changing functionality.

File: packages/dynamic-flows/src/step/layoutStep/LayoutStep.js (hunk 2 above)
Comment (JulianG-TW) [reply packages/dynamic-flows/src/step/layoutStep/LayoutStep.js:9]: I wanted to convey that they're the options for step conversion (something that I hope to get rid of some day) and not the options for "getting components".

Comment (JulianG-TW) [reply packages/dynamic-flows/src/step/layoutStep/LayoutStep.js:9]: oh, I see. in thiscontext.... yes. it could be called just options, I suppose.

Comment (JulianG-TW) [reply packages/dynamic-flows/src/step/layoutStep/LayoutStep.js]: renamed!

**Pull request merged**